### PR TITLE
Orders

### PR DESCRIPTION
**Pull request by andrzejkukuryk**

### CODE REVIEW

File: src/scenes/creator/final/cutlery.tsx [hunk 1]
```diff
@@ -2,16 +2,17 @@ import React from "react";
...
-import { cutlerySelector } from "../../../store/cutlery/selectors";
-import { updateAddCutlery } from "../../../store/cutlery/cutlerySlice";
 import { Checkbox } from "../../../components/checkbox";
+import { addToOrderSelector } from "../../../store/addToOrder/selectors";
+import { addAdditionally } from "../../../store/addToOrder/addToOrderSlice";
 
 export function Cutlery() {
   const dispatch = useDispatch();
-  const cutlery = useSelector(cutlerySelector);
+  const TYPE_CUTLERY = "CUTLERY";
```
Comment (anna-morawska) [inline src/scenes/creator/final/cutlery.tsx]: try to define constants outside of the component body

File: src/scenes/creator/final/final.tsx [hunk 2]
```diff
@@ -6,8 +6,17 @@ import { Name } from "./name";
...
 import { AppButton } from "../../../components/appButton";
+import { useDispatch } from "react-redux";
+import { createOrder } from "../../../store/orders/ordersSlice";
 
 export function Final() {
+  const dispatch = useDispatch();
+
+  const handleClickOrder = () => [
+    //@ts-ignore
+    dispatch(createOrder()),
```
Comment (anna-morawska) [inline src/scenes/creator/final/final.tsx]: https://redux-toolkit.js.org/usage/usage-with-typescript#getting-the-dispatch-type

maybe try this, it  should fix the TS error

File: src/scenes/creator/final/name.tsx [hunk 3]
```diff
@@ -1,28 +1,93 @@
...
 export function Name() {
   const dispatch = useDispatch();
 
-  const paniniName = useSelector(nameSelector);
+  const addCheese = useSelector(addCheeseSelector);
+  const currentCheese = useSelector(cheesesSelector);
+  const cheese = addCheese ? currentCheese : [];
+
+  const addMeat = useSelector(addMeatSelector);
+  const currentMeat = useSelector(meatsSelector);
```
Comment (anna-morawska) [inline src/scenes/creator/final/name.tsx]: `const meat = addMeat ? currentMeat : [];`

here, and with other ingredients you can try to move this logic to a seperate selector. 
it's good to keep "the buisness logic" in one place, so in this case in redux related files. 
You can combine selectors addMeatSelector and meatsSelector 
 
 https://redux.js.org/usage/deriving-data-selectors#createselector-overview

File: src/scenes/creator/final/name.tsx [hunk 4]
```diff
@@ -1,28 +1,93 @@
...
+        }
+      }
+    });
+
+    const newName =
+      mainIngredients.length === 2
+        ? `${mainIngredients[0]} & ${mainIngredients[1]} SANDWICH`
+        : mainIngredients.length === 1
+        ? `${mainIngredients[0]} SANDWICH`
+        : "DRY BREAD FOR THE HORSE";
```
Comment (anna-morawska) [inline src/scenes/creator/final/name.tsx:51]: 🐴

File: src/scenes/orders/dropdownOrders.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  const orders = useSelector(ordersSelector);
+  const currentOrderIndex = useSelector(currentOrderIndexSelector);
+
+  const dropdownOptions: DropdownInfo[] = orders.map((order) => {
+    return { value: order.id, label: order.name };
+  });
+
+  const handleChange = (selectedOption: SingleValue<DropdownInfo>) => {
+    if (selectedOption !== null) {
+      const currentOrder = orders.find(
```
Comment (anna-morawska) [inline src/scenes/orders/dropdownOrders.tsx]: maybe use straight away findIndex?

File: src/scenes/creator/final/cutlery.tsx (hunk 1 above)
Comment (andrzejkukuryk) [reply src/scenes/creator/final/cutlery.tsx]: Fixed

File: src/scenes/creator/final/name.tsx (hunk 3 above)
Comment (andrzejkukuryk) [reply src/scenes/creator/final/name.tsx]: I could finally use createSelector function :) It's done!

File: src/scenes/orders/dropdownOrders.tsx (hunk 5 above)
Comment (andrzejkukuryk) [reply src/scenes/orders/dropdownOrders.tsx]: I didn't know that method ;) It simplifies a lot. Thank you!

File: src/scenes/creator/final/final.tsx (hunk 2 above)
Comment (andrzejkukuryk) [reply src/scenes/creator/final/final.tsx]: Fixed.

**Pull request merged**